hazelcast/src/main/java/com/hazelcast/instance/impl/ClusterTopologyIntent.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.instance.impl;
+
+/**
+ * This class represents the estimated intent of topology changes
+ * performed in a managed runtime context (kubernetes) that may affect the cluster.
+ *
+ * @see ClusterTopologyIntentTracker
+ */
+public enum ClusterTopologyIntent {
```

Comment:
```suggestion
public enum ManagedContextIntent {
```
I'm playing with possible naming suggestions to best express the context here, changes are up to you and can be ok to see these minorish.